install/src/Install.hs

```diff
@@ -0,0 +1,144 @@
+
+  -- used for stack-based targets
+  hieVersions <- getHieVersions
+
+  putStrLn $ "run from: " ++ buildSystem
+
+  shakeArgs shakeOptions { shakeFiles = "_build" } $ do
+    want ["short-help"]
+    -- general purpose targets
+    phony "submodules"  updateSubmodules
+    phony "cabal"       installCabal
+    phony "short-help"  shortHelpMessage
+    phony "all"         shortHelpMessage
+    phony "help"        helpMessage
+    phony "check-stack" checkStack
+    phony "check-cabal" checkCabal
+
+    phony "cabal-ghcs" $ do
+      let
+        msg =
+          "Found the following GHC paths: \n"
+            ++ unlines
+                 (map (\(version, path) -> "ghc-" ++ version ++ ": " ++ path)
+                      ghcPaths
+                 )
+      printInStars msg
+
+    -- default-targets
+    phony "build" $ need [buildSystem ++ "-build"]
+    phony "build-all" $ need [buildSystem ++ "-build-all"]
+    phony "build-data" $ need [buildSystem ++ "-build-data"]
+    forM_
+      hieVersions
+      (\version ->
+        phony ("hie-" ++ version) $ need [buildSystem ++ "-hie-" ++ version]
+      )
+
+    -- stack specific targets
+    phony "stack-build"     (need (reverse $ map ("hie-" ++) hieVersions))
+    phony "stack-build-all" (need ["build-data", "build"])
+    phony "stack-build-data" $ do
+      need ["submodules"]
+      need ["check-stack"]
+      stackBuildData
+    forM_
+      hieVersions
+      (\version -> phony ("stack-hie-" ++ version) $ do
+        need ["submodules"]
+        need ["check-stack"]
+        need ["cabal"]
+        stackBuildHie version
+        stackInstallHie version
+      )
+
+    -- cabal specific targets
+    phony "cabal-build"     (need (map ("cabal-hie-" ++) ghcVersions))
+    phony "cabal-build-all" (need ["cabal-build-data", "cabal-build"])
+    phony "cabal-build-data" $ do
+      need ["submodules"]
+      need ["cabal"]
+      cabalBuildData
+    forM_
+      ghcVersions
+      (\version -> phony ("cabal-hie-" ++ version) $ do
+        validateCabalNewInstallIsSupported
+        need ["submodules"]
+        need ["cabal"]
+        cabalBuildHie version
+        cabalInstallHie version
+      )
+
+    -- macos specific targets
+    phony "icu-macos-fix"
+          (need ["icu-macos-fix-install"] >> need ["icu-macos-fix-build"])
+    phony "icu-macos-fix-install" (command_ [] "brew" ["install", "icu4c"])
+    phony "icu-macos-fix-build" $ mapM_ buildIcuMacosFix hieVersions
+
+
+buildIcuMacosFix :: VersionNumber -> Action ()
+buildIcuMacosFix version = execStackWithGhc_
+  version
+  [ "build"
+  , "text-icu"
+  , "--extra-lib-dirs=/usr/local/opt/icu4c/lib"
+  , "--extra-include-dirs=/usr/local/opt/icu4c/include"
+  ]
+
+-- | update the submodules that the project is in the state as required by the `stack.yaml` files
+updateSubmodules :: Action ()
+updateSubmodules = do
+  command_ [] "git" ["submodule", "sync"]
+  command_ [] "git" ["submodule", "update", "--init"]
```

Comment:
Is that a fix for #1218? Is that tested? Is that intentional? Should mention it in the PR.

Reply:
This is intentional. Tests have been successful.

Reply:
Then also mention it in the pr!